File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/service/model/user/ExamSlammrUserProfile.java
package uk.co.claritysoftware.exam.slammr.webapp.service.model.user;

import java.time.ZonedDateTime;
import java.util.Set;

import lombok.Builder;
import lombok.Value;

/**
 * Pojo defining an Exam Slammr User Profile
 */
@Value
@Builder(toBuilder = true)
public class ExamSlammrUserProfile {

	private String id;

	private String compositeUserId;

	private String firstname;

	private String surname;

	private String nickname;

	private String email;

	private String profilePictureUrl;

	private Set<String> roles;

	private ZonedDateTime registrationDateTime;

	private ZonedDateTime lastLogonDateTime;

}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/service/model/user/ExamSlammrUserProfile.kt
package uk.co.claritysoftware.exam.slammr.webapp.service.model.user

import java.time.ZonedDateTime

/**
 * Pojo defining an Exam Slammr User Profile
 */
data class ExamSlammrUserProfile(
		val id: String? = null,

		val compositeUserId: String? = null,

		val firstname: String? = null,

		val surname: String? = null,

		val nickname: String? = null,

		val email: String? = null,

		val profilePictureUrl: String? = null,

		val roles: Set<String>? = null,

		val registrationDateTime: ZonedDateTime? = null,

		val lastLogonDateTime: ZonedDateTime? = null
)